base iface: Provide permanent-mac-address when querying

cathay4t opened the pull request:
No test required as this is query only property and CI(container) has no
real NIC with this property set.

Manually tested in VM `virtio_net` NIC.

liangwen12year commented:
The CI is broken by the patch, when serializing the net_state, `permanent-mac-address` is not recognized or interpreted as the unknown field, please check it.

cathay4t commented:
> The CI is broken by the patch, when serializing the net_state, `permanent-mac-address` is not recognized or interpreted as the unknown field, please check it.

Fixed. Thanks!

liangwen12year commented on rust/src/lib/ifaces/base.rs:
```diff
@@ -50,11 +50,10 @@ pub struct BaseInterface {
     /// Serialize and deserialize to/from `mac-address`.
     pub mac_address: Option<String>,
-    #[serde(skip)]
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Since it is a query only property, how about adding `skip_deserializing` ?

cathay4t replied:
`skip_deserializing` will cause `nmstatectl fmt` lose it.

cathay4t committed:
base iface: Provide permanent-mac-address when querying

No test required as this is query only property and CI(container) has no
real NIC with this property set.

Manually tested in VM `virtio_net` NIC.

Signed-off-by: Gris Ge <fge@redhat.com>
